rust/operator/src/kafka_controller.rs

```diff
@@ -444,9 +454,7 @@ fn build_broker_role_serviceaccount(
             .with_recommended_labels(
                 kafka,
                 APP_NAME,
-                kafka
-                    .image_version()
-                    .context(KafkaVersionParseFailureSnafu)?,
+                &resolved_product_image.product_version,
```

Comment:
This change implies that we now one have different label values. Previously we had e.g. `1.4.1-stackable2.1.0`, now we have `1.4.1`. I'm not entirely sure if this causes problems, but in the `custom` product image variant the stackable version isn't known, so we can't add the "fully qualified" version.

Reply:
Had a short talk at arch meeting and decided to wait on the discussion regarding calling our product-images e.g. `1.4.1-release22.09`